### PR TITLE
Add LDAP verify cert option in configuration

### PR DESCRIPTION
**Pull request by stonezdj**
Add LDAP verify cert in configuration
https://github.com/vmware/harbor/issues/3285

### CODE REVIEW

File: src/common/utils/ldap/ldap.go [hunk 1]
```diff
@@ -346,7 +349,14 @@ func dialLDAP(ldapConfs models.LdapConf) (*goldap.Conn, error) {
...
 	case "ldaps":
-		ldap, err = goldap.DialTLS("tcp", hostport, &tls.Config{InsecureSkipVerify: true})
+		if ldapConfs.LdapVerifyCert {
+			log.Error("Connect ldaps with verified certificate!")
+			ldap, err = goldap.DialTLS("tcp", hostport, &tls.Config{ServerName: host, InsecureSkipVerify: false})
+		} else {
+			log.Errorf("Connect ldaps skip verified certificate!, ldapConfs=%v", ldapConfs)
+			ldap, err = goldap.DialTLS("tcp", hostport, &tls.Config{InsecureSkipVerify: true})
+		}
+
```
Comment (reasonerjt) [inline src/common/utils/ldap/ldap.go]: How about just `tls.Config{ServerName: host, InsecureSkipVerify: !ldapConfs.LdapVerifyCert}` ?

File: src/common/utils/ldap/ldap.go [hunk 2]
```diff
@@ -337,6 +338,8 @@ func dialLDAP(ldapConfs models.LdapConf) (*goldap.Conn, error) {
 	var ldap *goldap.Conn
 	splitLdapURL := strings.Split(ldapConfs.LdapURL, "://")
 	protocol, hostport := splitLdapURL[0], splitLdapURL[1]
+	splithostport := strings.Split(hostport, ":")
+	host := splithostport[0]
```
Comment (reasonerjt) [inline src/common/utils/ldap/ldap.go]: host := strings.Split(hostport, ":")[0]

File: src/ui/config/config.go [hunk 3]
```diff
@@ -183,6 +182,11 @@ func LDAP() (*models.LDAP, error) {
 	ldap.Filter = cfg[common.LDAPFilter].(string)
 	ldap.Scope = int(cfg[common.LDAPScope].(float64))
 	ldap.Timeout = int(cfg[common.LDAPTimeout].(float64))
+	if cfg[common.LDAPVerifyCert] != nil {
+		ldap.VerifyCert = cfg[common.LDAPVerifyCert].(bool)
+	} else {
+		ldap.VerifyCert = false
```
Comment (reasonerjt) [inline src/ui/config/config.go]: this is not necessary by default a boolean member will be false.

File: src/common/utils/ldap/ldap.go (hunk 1 above)
Comment (stonezdj) [reply src/common/utils/ldap/ldap.go]: Updated

File: src/ui/config/config.go (hunk 3 above)
Comment (stonezdj) [reply src/ui/config/config.go]: Updated

File: src/common/utils/ldap/ldap.go (hunk 2 above)
Comment (stonezdj) [reply src/common/utils/ldap/ldap.go]: Updated

**Pull request merged**